app/src/main/java/com/example/tour_a_versity_v3/MapsActivity.kt

```diff
@@ -127,36 +123,24 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback, GoogleMap.OnInfoWi
         )
 
     }
+
+    fun createCechMarker() {
+        val cechLat = 39.130315
+        val cechLng = -84.518680
+        val cechLatLng = LatLng(cechLat, cechLng)
+        cechMarker = map.addMarker(MarkerOptions()
+            .position(cechLatLng)
+            .title("CECH")
+            .snippet("College of Education, Criminal Justice, Human Services, and IT")
+        )
+    }
+
     override fun onInfoWindowClick(p0: Marker?) {
         if (p0 == tucMarker){
             val intent = Intent(this, TUC_Info::class.java)
             startActivity(intent)
         }
-        else if (p0 == cechMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == pavilionMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == lawMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == belgenMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == mcmickenMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == braunsteinMarker){
-            val intent = Intent(this, CECH_Info::class.java)
-            startActivity(intent)
-        }
-        else if (p0 == daapMarker){
+        else {
```

Comment:
Repeated code, there is no reason to add so many conditional branches that to the exact same thing.

Reply:
If tests are often a sign of object oriented design that is not optimized.  This is one of those cases.  There should not be a separate class for each building, because you'll need to update the program each time you add a building.